src/crimson/osd/backfill_state.cc

```diff
@@ -147,25 +147,12 @@ void BackfillState::Enqueuing::maybe_update_range()
       [&](const pg_log_entry_t& e) {
         logger().debug("maybe_update_range(lambda): updating from version {}",
                        e.version);
```

Comment:
I *think* it's safe to assume that if the projected log has any contents, get_projected_last_update == projected_log.back().version.  (It's not quite safe to assume the same for the actual log because split can leave holes at the end of the log.)